### ISSUE TITLE
Exception while trying to report linter executable not being found

### ISSUE THREAD

**Issue by FichteFoll** [SublimeLinter/SublimeLinter#403]
```
Traceback (most recent call last):
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 256, in on_post_save
    callback.on_post_save(v)
  File "C:\Users\gautamz07\AppData\Roaming\Sublime Text 3\Packages\SublimeLinter\sublimelinter.py", line 441, in on_post_save
    self.file_was_saved(view)
  File "C:\Users\gautamz07\AppData\Roaming\Sublime Text 3\Packages\SublimeLinter\sublimelinter.py", line 445, in file_was_saved
    syntax_changed = self.check_syntax(view)
  File "C:\Users\gautamz07\AppData\Roaming\Sublime Text 3\Packages\SublimeLinter\sublimelinter.py", line 205, in check_syntax
    Linter.assign(view, reset=True)
  File "C:\Users\gautamz07\AppData\Roaming\Sublime Text 3\Packages\SublimeLinter\lint\linter.py", line 728, in assign
    for name, linter_class in persist.linter_classes.items():
  File "./functools.py", line 251, in wrapper
  File "C:\Users\gautamz07\AppData\Roaming\Sublime Text 3\Packages\SublimeLinter\lint\linter.py", line 1588, in can_lint
    elif status is None:
UnboundLocalError: local variable 'executable' referenced before assignment
```

See https://forum.sublimetext.com/t/plugins-in-sublime-stop-working-occasionally/16547 for context.

**Comment by aristidesfl** [comment]
Same here:

```
Traceback (most recent call last):
  File "/Applications/Sublime Text.app/Contents/MacOS/sublime_plugin.py", line 389, in run_callback
    expr()
  File "/Applications/Sublime Text.app/Contents/MacOS/sublime_plugin.py", line 522, in <lambda>
    run_callback('on_activated', callback, lambda: callback.on_activated(v))
  File "/Users/user/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/sublimelinter.py", line 278, in on_activated
    self.check_syntax(view)
  File "/Users/user/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/sublimelinter.py", line 205, in check_syntax
    Linter.assign(view, reset=True)
  File "/Users/user/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/lint/linter.py", line 734, in assign
    if not linter_class.disabled and linter_class.can_lint(syntax):
  File "./python3.3/functools.py", line 251, in wrapper
  File "/Users/user/Library/Application Support/Sublime Text 3/Packages/SublimeLinter/lint/linter.py", line 1602, in can_lint
    status = 'WARNING: {} deactivated, cannot locate \'{}\''.format(cls.name, executable)
UnboundLocalError: local variable 'executable' referenced before assignment
```

**Comment by aristidesfl** [comment]
Btw, this is happening when I open `js` files.
The syntax highlight is to `JavaScript (Babel)`, and I have `SublimeLinter-contrib-eslint_d` installed. The same happens with sublime `SublimeLinter-contrib-eslint`.

```bash
$ which eslint_d
/usr/local/bin/eslint_d
```

In SublimeLinter User Setting I have: 

```json
        "syntax_map": {
            "JavaScript (Babel)": "javascript",
            "html (django)": "html",
            "html (rails)": "html",
            "html 5": "html",
            "php": "html",
            "python django": "python"
        },
        "paths": {
            "linux": [],
            "osx": [
                "/usr/local/bin"
            ],
            "windows": []
        },
```

**Comment by braver** [comment]
We have been making [many changes](https://github.com/SublimeLinter/SublimeLinter/blob/master/messages/4.0.0-rc.1.txt) to SublimeLinter, some of which also should have fixed this particular problem. It would be really nice if you could [try the beta](https://github.com/SublimeLinter/SublimeLinter/tree/15bb4a2e2024a623f60bb444eadd5790764fdfb9#sublimelinter-4-beta) and let us know if that fixes the problem for you.

**Comment by FichteFoll** [comment]
I haven't been able to reproduce this in quite a while, iirc. With the architectural changes for SL4, I expect to not see this issue again and if I do, I'll open a new issue with an updated traceback.

**Comment by FichteFoll** [comment]
(Actually, it makes sense that I don't remember seeing this if it wasn't even my traceback, huh. I guess someone should just take a look at the specified code location to see if a branch leading to this exception is still possible.)

**Comment by braver** [comment]
This was fixed in #491 actually.
